src/rawposix/src/sys_calls.rs

```diff
@@ -0,0 +1,1117 @@
+        false,
+        0,
+    )
+    .unwrap();
+    // STDERR
+    fdtables::get_specific_virtual_fd(
+        0,
+        STDERR_FILENO as u64,
+        FDKIND_KERNEL,
+        STDERR_FILENO as u64,
+        false,
+        0,
+    )
+    .unwrap();
+
+    //init cage is its own parent
+    let initcage = Cage {
+        cageid: 1,
+        cwd: RwLock::new(Arc::new(PathBuf::from("/"))),
+        parent: 1,
+        rev_shm: Mutex::new(Vec::new()),
+        main_threadid: RwLock::new(0),
+        interval_timer: IntervalTimer::new(1),
+        epoch_handler: DashMap::new(),
+        signalhandler: DashMap::new(),
+        pending_signals: RwLock::new(vec![]),
+        sigset: AtomicU64::new(0),
+        zombies: RwLock::new(vec![]),
+        child_num: AtomicU64::new(0),
+        vmmap: RwLock::new(Vmmap::new()),
+    };
+
+    // Add cage to cagetable
+    add_cage(
+        1, // cageid
+        initcage,
+    );
+
+    fdtables::init_empty_cage(1);
+    // Set the first 3 fd to STDIN / STDOUT / STDERR
+    // STDIN
+    fdtables::get_specific_virtual_fd(
+        1,
+        STDIN_FILENO as u64,
+        FDKIND_KERNEL,
+        STDIN_FILENO as u64,
+        false,
+        0,
+    )
+    .unwrap();
+    // STDOUT
+    fdtables::get_specific_virtual_fd(
+        1,
+        STDOUT_FILENO as u64,
+        FDKIND_KERNEL,
+        STDOUT_FILENO as u64,
+        false,
+        0,
+    )
+    .unwrap();
+    // STDERR
+    fdtables::get_specific_virtual_fd(
+        1,
+        STDERR_FILENO as u64,
+        FDKIND_KERNEL,
+        STDERR_FILENO as u64,
+        false,
+        0,
+    )
+    .unwrap();
+}
+
+pub fn rawposix_shutdown() {
+    let exitvec = cagetable_clear();
+
+    for cageid in exitvec {
+        exit_syscall(
+            cageid as u64,       // target cageid
+            EXIT_SUCCESS as u64, // status arg
+            cageid as u64,       // status arg's cageid
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+        );
+    }
+}
+
+/// Reference to Linux: https://man7.org/linux/man-pages/man3/exec.3.html
+///
+/// In our implementation, WASM is responsible for handling functionalities such as loading and executing
+/// the new program, preserving process attributes, and resetting memory and the stack.
+///
+/// In RawPOSIX, the focus is on memory management inheritance and resource cleanup and release. Specifically,
+/// RawPOSIX handles tasks such as clearing memory mappings, resetting shared memory, managing file descriptors
+/// (closing or inheriting them based on the `should_cloexec` flag in fdtable), resetting semaphores, and
+/// managing process attributes and threads (terminating unnecessary threads). This allows us to fully implement
+/// the exec functionality while aligning with POSIX standards. Cage fields remained in exec():
+/// cageid, cwd, parent, interval_timer
+pub fn exec_syscall(
```

Comment:
its kind of weird that this syscall is all the way at the bottom of this file, probably should be next to fork

Reply:
also similar to fork would be good to ref to wasmtime code